src/main/src/main/java/org/geoserver/ows/LanguageURLMangler.java

```diff
@@ -0,0 +1,55 @@
+/* (c) 2021 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.ows;
+
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+
+/** Mangles service URL's based on the AcceptLanguages and Languages parameters */
+public class LanguageURLMangler implements URLMangler {
+
+    public static final String ACCEPT_LANGUAGES = "AcceptLanguages";
+    public static final String LANGUAGE = "Language";
+    private String languageParameter = "";
+    private String acceptLanguagesParameter = "";
+
+    @Override
+    public void mangleURL(
+            StringBuilder baseURL, StringBuilder path, Map<String, String> kvp, URLType type) {
+        if (Dispatcher.REQUEST.get() == null || !type.equals(URLType.SERVICE)) return;
+
+        String language = "";
+        HttpServletRequest httpRequest = Dispatcher.REQUEST.get().getHttpRequest();
+        Enumeration<String> parameterNames = httpRequest.getParameterNames();
+        Collections.list(parameterNames)
+                .forEach(
+                        parameter -> {
+                            if (parameter.equalsIgnoreCase(LANGUAGE)) {
+                                languageParameter = parameter;
+                            } else if (parameter.equalsIgnoreCase(ACCEPT_LANGUAGES)) {
+                                acceptLanguagesParameter = parameter;
+                            }
+                        });
+
+        String acceptLanguages = httpRequest.getParameter(acceptLanguagesParameter);
```

Comment:
The javadoc does not say anything about the parameter name null-ness. I'm worried container specific implementations of HttpServletRequest would throw a NPE if the provided name is null (it is not a very sensible thing to do anyways).

Javadoc here: https://docs.oracle.com/javaee/6/api/javax/servlet/ServletRequest.html#getParameter(java.lang.String)

Reply:
Added null check for acceptLanguages and languageParameter.